asynctest/case.py

```diff
@@ -306,19 +316,27 @@ def debug(self):
             function(*args, **kwargs)
 
     def _run_test_method(self, method):
+        fail_checks = self._get_fail_checks(method)
         # If the method is a coroutine or returns a coroutine, run it on the
         # loop
         result = method()
         if asyncio.iscoroutine(result):
             self.loop.run_until_complete(result)
-            return
-
-        if (getattr(self.__class__, "__asynctest_ignore_loop__", False) or
-                getattr(method, "__asynctest_ignore_loop__", False)):
-            return
-
-        if not self.loop.__asynctest_ran:
-            self.fail("Loop did not run during the test")
+        else:
+            if fail_checks['unused_loop']:
+                if not self.loop.__asynctest_ran:
+                    self.fail("Loop did not run during the test")
+            if fail_checks['active_handles']:
```

Comment:
the "unused loop" test is useless when the test method is a coroutine (the loop will run the coroutine), but "active handles" can (and should) be tested in both cases (coroutine or function).